src/Sarif.Multitool/ResultMatchSetCommand.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis.Sarif.Baseline.ResultMatching;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool
+{
+    internal class ResultMatchSetCommand : CommandBase
+    {
+        private readonly IFileSystem _fileSystem;
+
+        public ResultMatchSetCommand(IFileSystem fileSystem = null)
+        {
+            _fileSystem = fileSystem ?? new FileSystem();
+        }
+
+        public int Run(ResultMatchSetOptions options)
+        {
+            options.OutputFolderPath = options.OutputFolderPath ?? Path.Combine(options.FolderPath, "Out");
+
+            ISarifLogMatcher matcher = ResultMatchingBaselinerFactory.GetDefaultResultMatchingBaseliner();
+            Formatting formatting = (options.PrettyPrint ? Formatting.Indented : Formatting.None);
+
+            // Remove previous results
+            if (Directory.Exists(options.OutputFolderPath))
+            {
+                Directory.Delete(options.OutputFolderPath, true);
+            }
+
+            // Create output folder
+            Directory.CreateDirectory(options.OutputFolderPath);
+
+            string previousFileName = "";
+            string previousGroup = "";
+            SarifLog previousLog = null, currentLog = null;
+
+            foreach (string filePath in Directory.GetFiles(options.FolderPath, "*.sarif"))
+            {
+                string fileName = Path.GetFileName(filePath);
+                string currentGroup = GetGroupName(fileName);
+
+                try
+                {
+                    currentLog = ReadSarifFile<SarifLog>(_fileSystem, filePath);
+
+                    // Compare each log with the previous one in the same group
+                    if (currentGroup.Equals(previousGroup) && currentLog?.Runs[0].Results.Count != 0 && previousLog?.Runs[0].Results.Count != 0)
```

Comment:
>currentLog?.Runs[0].Results.Count != 0 [](start = 62, length = 38)

Is this test right? If the previous run had results but the current one doesn't, don't we want to produce a log where all the results are marked "absent"? If the current run has results but the previous one didn't, don't we want to produce a log where all the results are marked "new"?

Reply:
(Of course if this is the first run, we want to leave it alone.)

---
In reply to: [296047688](https://github.com/microsoft/sarif-sdk/pull/1539#discussion_r296047688) [](ancestors = 296047688)